File: spring-boot-tipps-reactive/pnia/src/main/java/io/github/jlmc/reactive/domain/services/blacklist/BlackListService.java
package io.github.jlmc.reactive.domain.services.blacklist;

import org.springframework.stereotype.Service;

import java.util.Collection;
import java.util.Objects;
import java.util.Set;
import java.util.stream.Collectors;

@Service
public class BlackListService {

    private final Set<String> blackList = Set.of("true", "false", "TRUE", "FALSE");

    public Collection<String> getForbiddenInputs(Collection<String> inputs) {
        return inputs.stream().filter(it -> Objects.isNull(it) || blackList.contains(it)).collect(Collectors.toSet());
    }

}

File: spring-boot-tipps-reactive/pnia-kotlin/src/main/kotlin/io/github/jlmc/pniakotlin/domain/services/blacklist/BlackListService.kt
package io.github.jlmc.pniakotlin.domain.services.blacklist

import org.springframework.stereotype.Service

@Service
class BlackListService {

    private val blackList = setOf("true", "false", "TRUE", "FALSE")

    private fun getForbiddenInputs(inputs: Collection<String?>): List<String?> {
        return inputs.filter { it.isNullOrBlank() || blackList.contains(it) }
    }

    fun validate(inputs: Collection<String?>) {
        val forbiddenInputs = getForbiddenInputs(inputs)
        if (forbiddenInputs.isNotEmpty()) {
            throw ForbiddenInputValuesException(forbiddenInputs)
        }
    }

}
